### PR TITLE
Added block menu in notification list

### PR DESCRIPTION
**Pull request by erditkurteshiSQA**
Fixes [#776](https://github.com/near/near-discovery/issues/776).

Added block menu on notification list and also confirmation modal.

 Related with: https://github.com/near/near-discovery/pull/820.

block menu

image

### CODE REVIEW

File: src/NearOrg/Notifications/Notification.jsx [hunk 1]
```diff
@@ -14,12 +14,15 @@ const Notification = styled.div`
   }
 `;
 
-const Content = styled.div`
+const Content = styled.a`
```
Comment (shelegdmitriy) [inline src/NearOrg/Notifications/Notification.jsx:17]:
```suggestion
const Content = styled("Link")`
```

@erditkurteshiSQA check out this [discussion](https://github.com/near/near-discovery-components/discussions/498) for the reference

File: src/NearOrg/Notifications/Notification.jsx [hunk 2]
```diff
@@ -322,5 +357,35 @@ return (
...
     </Content>
+    <Widget
+          src="${REPL_ACCOUNT}/widget/DIG.DropdownMenu"
+          props={{
+            trigger: <i className="ph-bold ph-dots-three" />,
+            items: buildMenu(),
+          }}
+        />
+        {isModalOpen && (
+        <div className="modal" style={{ display: 'block', backgroundColor: 'rgba(0, 0, 0, 0.6)' }}>
```
Comment (shelegdmitriy) [inline src/NearOrg/Notifications/Notification.jsx:368]: For the modal we have a [DIG.Dialog](https://near.org/near/widget/ComponentDetailsPage?src=near/widget/DIG.Dialog&tab=about) component

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
@erditkurteshiSQA sorry for reviewing the draft version of this PR but I left a couple of comments

File: src/NearOrg/Notifications/Notification.jsx (hunk 1 above)
Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/Notification.jsx]: @shelegdmitriy This has been addressed in the latest commits. Thank you for the review.

File: src/NearOrg/Notifications/Notification.jsx (hunk 2 above)
Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/Notification.jsx]: This has been also addressed in the latest commit. Thank you for the review.

File: src/NearOrg/Notifications/Notification.jsx [hunk 3]
```diff
@@ -130,6 +133,7 @@ const path = item.path || "";
 
 // Build notification
 let { blockHeight, accountId } = props;
+let { manageNotification } = props;
```
Comment (shelegdmitriy) [inline src/NearOrg/Notifications/Notification.jsx:136]:
```suggestion
let { blockHeight, accountId, manageNotification } = props;
```

File: src/NearOrg/Notifications/NotificationsList.jsx [hunk 4]
```diff
@@ -16,6 +16,8 @@ const index = {
   },
 };
 
+let { manageNotification } = props;
```
Comment (shelegdmitriy) [inline src/NearOrg/Notifications/NotificationsList.jsx:19]: Let's move this to line [4](https://github.com/near/near-discovery-components/pull/520/files#diff-3665dbffa6eb0e091687824986cb9b62bdc720ceb6d9dccf8bdb38d59d67d9b2R4) so all the props will be at the same place at the top

File: src/NearOrg/Notifications/Notification.jsx [hunk 5]
```diff
@@ -322,5 +366,24 @@ return (
...
+      props={{
+        trigger: <i className="ph-bold ph-dots-three" />,
+        items: buildMenu(),
+      }}
+    />
+    <Widget
+      src="${REPL_ACCOUNT}/widget/DIG.Dialog"
+      props={{
+        type: "alert",
+        title: <modalContent />,
```
Comment (shelegdmitriy) [inline src/NearOrg/Notifications/Notification.jsx:380]: I think we can avoid this function and use this instead:
```suggestion
        title: `Do you want to stop receiving push notification for ${type}`,
```

**Review by shelegdmitriy — APPROVED** [review]
@erditkurteshiSQA LGTM! I just added a few comments as a cleanup.

File: src/NearOrg/Notifications/Notification.jsx (hunk 3 above)
Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/Notification.jsx]: This has been addressed in the latest commit. Thank you

File: src/NearOrg/Notifications/NotificationsList.jsx (hunk 4 above)
Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/NotificationsList.jsx]: This has been addressed in the latest commit. Thank you

File: src/NearOrg/Notifications/Notification.jsx (hunk 5 above)
Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/Notification.jsx]: This also has been addressed in the latest commit. Thank you

**Comment by erditkurteshiSQA** [comment]
> @erditkurteshiSQA LGTM! I just added a few comments as a cleanup.

Thank you @shelegdmitriy ,
Let's hold on merging until the PR: https://github.com/near/near-discovery/pull/820 is resolved. Once it's addressed, we can continue with the merge.

File: src/NearOrg/Notifications/Notification.jsx [hunk 6]
```diff
@@ -259,10 +262,46 @@ const iconType = {
...
+const buildMenu = () => {
+  return [
+    {
+      name: (
+        <>
+          <i
+            className="ph-bold ph-bell-simple-slash"
+            style={{ color: "#D95C4A" }}
+          />
+          <span style={{ color: "#D95C4A" }}>Block</span>
```
Comment (milenakrawczyk) [inline src/NearOrg/Notifications/Notification.jsx:276]: This should be visible only for users that enabled push notifications. It seems that checking whether a subscription endpoint exists should be enough to determine whether the push notifications have been enabled.

Comment (erditkurteshiSQA) [reply src/NearOrg/Notifications/Notification.jsx:276]: I added a condition only for users that have enabled push notifications. Should be fixed in the latest commit. Thank you Milena.

**Pull request merged**